bundles/mapping/mapmodule/event/TimeChangedEvent.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @class Oskari.mapframework.event.common.TimeChangedEvent
+ */
+Oskari.clazz.define('Oskari.mapframework.event.common.TimeChangedEvent',
+    /**
+     * @static @method create called automatically on construction
+     */
+    function (dateTime) {
+        this._dateTime = dateTime;
+        this._dateObject = new Date(dateTime);
+    }, {
+        /** @static @property __name event name */
+        __name: 'TimeChangedEvent',
+
+        /**
+         * @method getName
+         * @return {String} event name
+         */
+        getName: function () {
+            return this.__name;
+        },
+
+        getDateTime: function () {
+            return this._dateTime;
+        },
+
+        getDateObject: function () {
+            return this._dateObject;
+        },
+
+        /**
+         * @method getDate
+         * @return {String} date formatted as D/M
+         */
+        getDate: function () {
+            const date = this.getDateObject();
+            return `${date.getDate()}/${date.getMonth() + 1}`;
+        },
+
+        /**
+         * @method getTime
+         * @return {String} time formatted as HH:mm
+         */
+        getTime: function () {
+            const date = this.getDateObject();
+            return `${date.getHours()}:${date.getMinutes()}`;
+        }
```

Comment:
Use moment js here instead for formatting?

Reply:
Since we already have moment.js as dependency sure you could use it. However I would move the code to where the event is sent and just have date and time passed in constructor (removing the getDateObject() and getDateTime() from the event).

Reply:
Add getParams() function to serialize this for RPC: https://oskari.org/guides/rpc-events